metrics-operator/api/v1alpha3/analysisdefinition_types.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AnalysisDefinitionSpec defines the desired state of AnalysisDefinition
+type AnalysisDefinitionSpec struct {
+	// Objectives defines a list of objectives for evaluation
+	Objectives []Objective `json:"objectives,omitempty"`
+	// TotalScore defines the required score for an evaluation to be successful
+	TotalScore Score `json:"totalScore"`
+}
+
+// Score defines the required score for an evaluation to be successful
+type Score struct {
```

Comment:
Nit, because most likely in the computation of the final score, we will use a score variable for each objective and could be confusing
```suggestion
type TotalScore struct {
```